components/user/Tabs/SettingsContent.tsx

```diff
@@ -1,59 +1,237 @@
-import React from 'react'
+import { useState, useEffect, useCallback } from 'react'
 
-import * as API from 'apiClient'
+import Note from 'components/Form/Note'
+import { FieldError } from 'components/Form/FieldStatus'
 import Select from 'components/Form/Select'
 import TextField from 'components/Form/TextField'
+import Button from 'components/Button'
+import Loader from 'components/Loader'
+// import Debug from 'components/Debug'
+
 import { useField } from 'hooks/useForm'
 import { FIELDS } from 'pages/signup'
 
+import { scrollUp } from 'utils/scroll'
+import { UNSET } from 'utils/forms'
+import { useQueriedToast } from 'hooks/useToast'
+
+import * as API from 'apiClient'
+import { TabType } from './index'
+
 type Props = {
-  authedUser: API.ApiUserMetadata
+  anyBlocksMined: boolean
+  user: API.ApiUser | null
+  authedUser: API.ApiUserMetadata | null
+  toast: ReturnType<typeof useQueriedToast>
+  reloadUser: () => void
+  onTabChange: (tab: TabType) => unknown
 }
 
-const getSocialChoice = (
-  authedUser: API.ApiUserMetadata
-): { choice: string; value: string } => {
-  if (authedUser.discord) {
-    return { choice: 'discord', value: authedUser.discord }
-  } else if (authedUser.telegram) {
-    return { choice: 'telegram', value: authedUser.telegram }
-  }
-  return { choice: '', value: '' }
+const EDITABLE_FIELDS = {
+  email: { ...FIELDS.email, validation: () => true, touched: true },
+  graffiti: { ...FIELDS.graffiti },
+  country: { ...FIELDS.country, useDefault: false },
+  discord: {
+    ...FIELDS.social,
+    required: false,
+    options: undefined,
+    id: 'discord',
+    label: 'Discord',
+    placeholder: 'Your Discord username',
+    isRadioed: false,
+    validation: () => true,
+  },
+  telegram: {
+    ...FIELDS.social,
+    required: false,
+    options: undefined,
+    id: 'telegram',
+    label: 'Telegram',
+    placeholder: 'Your Telegram username',
+    isRadioed: false,
+    validation: () => true,
+  },
 }
 
-export default function SettingsContent({ authedUser }: Props) {
-  const $email = useField(FIELDS.email)
-  const $graffiti = useField(FIELDS.graffiti)
-  const $social = useField(FIELDS.social)
-  const $country = useField(FIELDS.country)
+export default function SettingsContent({
+  anyBlocksMined,
+  user,
+  authedUser,
+  toast,
+  reloadUser,
+  onTabChange,
+}: Props) {
+  const [$error, $setError] = useState<string>(UNSET)
+  const [$userData, $setUserData] = useState<API.ApiUserMetadata | null>(
+    authedUser
+  )
+
+  const {
+    email: _email = UNSET,
+    graffiti: _graffiti = UNSET,
+    discord: _discord = UNSET,
+    telegram: _telegram = UNSET,
+    country_code: _cc = UNSET,
```

Comment:
This name can be expanded out to `_country_code` to match the style of the other variable names